share/lib/python/neuron/tests/utils/checkresult.py

```diff
@@ -48,33 +49,29 @@ def equal(a, b):
                 assert type(a) == type(b)
                 if type(a) in (float, int):
                     match = math.isclose(a, b, rel_tol=tol)
-                    if not match:
-                        print(a, b, "diff", abs(a - b) / max(abs(a), abs(b)), ">", tol)
+                    assert (
+                        match
+                    ), f"{a}, {b}, diff = {abs(a - b) / max(abs(a), abs(b))} > {tol}"
                     return match
                 elif type(a) == str:
                     match = a == b
-                    if not match:
-                        print("strdiff", a, b)
+                    assert match, f"strdiff: {a} != {b}"
                     return match
                 elif type(a) == list:
                     # List comprehension avoids short-circuit, so the "diff"
                     # message just above is printed for all elements
-                    return all([equal(aa, bb) for aa, bb in zip(a, b)])
+                    return all(equal(aa, bb) for aa, bb in zip(a, b))
                 elif type(a) == dict:
                     assert a.keys() == b.keys()
-                    return all([equal(a[k], b[k]) for k in a.keys()])
+                    return all(equal(a[k], b[k]) for k in a.keys())
                 raise Exception(
                     "Don't know how to compare objects of type " + str(type(a))
                 )
 
             match = equal(value, self.d[key])
-            if not match:
-                print(key, "difference")
-                print("std = ", self.d[key])
-                print("val = ", value)
-            assert match
```

Comment:
Prior to the assert, some information is printed that, I think, is helpful for debugging (key, std, val). Won't this be lost if you assert within def equal?  Do you think this prints too much information that you avoid by introducing two asserts within def equal?